### PR TITLE
cliccl: remove the debug backup commands

### PR DESCRIPTION
**Pull request by stevendanna**
These commands have been experimental and undocumented since their original development. Many of the commands provide the same functionality as "SHOW BACKUP".  The export command provides unique functionality, but it is hard to use correctly.

Further, leaving these in the code has forced us to maintain APIs in the backupccl code that we would have otherwised removed or refactored.

Release justification: Low risk removal of unsupported functionality.

Release note (cli change): The previously-experimental and undocumented `debug backup` has been removed.

### CODE REVIEW

File: pkg/ccl/cliccl/debug_backup.go [hunk 1]
```diff
@@ -1,851 +0,0 @@
...
-		)
-	}
-
-	endTime, err := evalAsOfTimestamp(debugBackupArgs.readTime, manifests)
-	if err != nil {
-		return errors.Wrapf(err, "eval as of timestamp %s", debugBackupArgs.readTime)
-	}
-
-	codec := keys.TODOSQLCodec
-	entry, err := backupccl.MakeBackupTableEntry(
```
Comment (dt) [inline pkg/ccl/cliccl/debug_backup.go:496]: I think this function can go from backupccl now too, which is has been the "ugh, why am I maintaining this" edge lately.

**Review by dt — APPROVED** [review]

**Comment by stevendanna** [comment]
bors r=dt

**Comment by stevendanna** [comment]
bors r=dt

**Comment by stevendanna** [comment]
bors retry

**Pull request merged**